### ISSUE TITLE
[a11y]: NumberInput- JAWS Not Announcing 'Readonly' Property for Input

### ISSUE THREAD

**Issue by 2nikhiltom** [carbon-design-system/carbon#16530]
### Package

@carbon/react

### Package version

v1.56.0

### React version

18.02

### Automated testing tool and ruleset

Manual QA

### Assistive technology

JAWS

### Description

When the 'Read only' Boolean is set to 'True' from the playground (i.e when the `input` is made `readOnly`) ,JAWS does not announce 'readonly' along with the label of the input field.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-numberinput--playground

### Recommend Fix
Ensure JAWS should announce the "read only" property along with the label of the input box

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
